tools/dep_updaters/update-v8-patch.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh
+set -e
+# Shell script to update v8 patch update
+
+BASE_DIR=$(cd "$(dirname "$0")/../.." && pwd)
+DEPS_DIR="$BASE_DIR/deps"
+
+CURRENT_MAJOR_VERSION=$(grep "#define V8_MAJOR_VERSION" "$DEPS_DIR/v8/include/v8-version.h" | cut -d ' ' -f3)
+CURRENT_MINOR_VERSION=$(grep "#define V8_MINOR_VERSION" "$DEPS_DIR/v8/include/v8-version.h" | cut -d ' ' -f3)
+CURRENT_BUILD_VERSION=$(grep "#define V8_BUILD_NUMBER" "$DEPS_DIR/v8/include/v8-version.h" | cut -d ' ' -f3)
+CURRENT_PATCH_VERSION=$(grep "#define V8_PATCH_LEVEL" "$DEPS_DIR/v8/include/v8-version.h" | cut -d ' ' -f3)
+
+CURRENT_VERSION_NO_PATCH="$CURRENT_MAJOR_VERSION.$CURRENT_MINOR_VERSION.$CURRENT_BUILD_VERSION"
+
+NEW_VERSION=$(git -c 'versionsort.suffix=-' ls-remote --tags --sort='v:refname' https://chromium.googlesource.com/v8/v8 "$CURRENT_VERSION_NO_PATCH.*" | tail -n1 | cut -d '/' -f3)
+
+LATEST_PATCH_VERSION=$(echo "$NEW_VERSION" | cut -d '.' -f4)
+
+if [ -z "$LATEST_PATCH_VERSION" ]; then
+  echo "Skipped because latest patch is not available."
+  exit 0
+fi
+
+echo "Comparing current patch version $CURRENT_PATCH_VERSION with latest patch $LATEST_PATCH_VERSION"
+
+if [ "$CURRENT_PATCH_VERSION" = "$LATEST_PATCH_VERSION" ]; then
+  echo "Skipped because v8 is on the latest version."
+  exit 0
+fi
+
+NEW_VERSION="$CURRENT_VERSION_NO_PATCH.$LATEST_PATCH_VERSION"
+
+CURRENT_VERSION="$CURRENT_VERSION_NO_PATCH.$CURRENT_PATCH_VERSION"
+
+WORKSPACE=$(mktemp -d 2> /dev/null || mktemp -d -t 'tmp')
+
+cleanup () {
+  EXIT_CODE=$?
+  [ -d "$WORKSPACE" ] && rm -rf "$WORKSPACE"
+  exit $EXIT_CODE
+}
+
+trap cleanup INT TERM EXIT
+
+cd "$WORKSPACE"
+
+echo "Cloning v8 repository..."
+
+git clone https://chromium.googlesource.com/v8/v8
```

Comment:
Perhaps something like this?:

```suggestion
git clone --depth=1 --branch $NEW_VERSION https://chromium.googlesource.com/v8/v8
```

Then use diff to generate a patch between `$DEPS_DIR/v8` and the checked out `v8/`?